tests/unit/test_endpoint_provider.py

```diff
@@ -544,3 +553,141 @@ def test_deeply_nested(self):
             ]
         }
         self.assertDictEqual(expected_rules_file, rules_file_dict_read)
+
+
+def provider_with_rules_factory(rules: Dict[str, str]):
```

Comment:
We now have multiple ways of creating customized test Provider Charms in our unit tests

On one hand we have two four line methods that use alert rules from file 
- `class CustomizableEndpointProviderCharm`
- `def customize_endpoint_provider`

On the other hand we have 
- The `helper.py` module that offers the `TempFolderSandbox` which writes alert rules (taken from this unittest script) into files on disk
- The `def provider_with_rules_factory()` method that uses `TempFolderSandbox` class to create alert rules and instantiate a Provider charm.

At least within one charm's code base should we be consistent in the way we write the unit tests ? Do you think there is value in doing this ? Would it make our code easier to understand ?

Reply:
Given the paucity of time, do you think it makes sense to make the smallest simplest change that will fix the bug, check it is fixed and deliver ? Would it make more sense to move the discussion to another refactoring ticket. ?

Reply:
Feel free to create a refactoring ticket!